src/fetch.js

```diff
@@ -43,14 +43,22 @@ export function runFetch ({
   retry = false,
   url
 }, state) {
+  const headers = {
+    ...createAuthorizationHeader(state),
+    ...createContentHeader(options.body),
+    ...(options.headers || {})
+  }
+
+  // allow removing generated headers by specifiying { header: null } in options.headers
+  // do this in two steps because otherwise we're modifying the object as we're iterating over it
+  const nullKeys =
```

Comment:
Let's replace this with something that doesn't include `delete`. Possibly using `lodash/transform`

```js
...
headers: transform(headers, (result, value, key) => {
  if (value != null) result[key] = value
}, {})
...
```